docs/docs/developers/contracts/writing_contracts/storage/notes.md

```diff
@@ -0,0 +1,78 @@
+- [Transaction lifecycle (simple diagram)](../../../../learn/concepts/transactions#simple-example-of-the-private-transaction-lifecycle)
+- [Public and Private state](../../../../learn/concepts/hybrid_state/main)
+
+Outside of this there are many more details to discover that will be skipped over for now:
+- Private and public contexts
+- Encryption keys and events
+- Oracle's role in using notes
+- Value Serialization/Deserialization
+
+This explainer will go through some examples that touch on what has been covered [here](../../../../learn/concepts/storage/notes).
+
+#### Some code context
+The way Aztec (`aztec-packages` monorepo) benefits from the Noir language is via three important components:
+- `Aztec-nr` - a Noir framework enabling contracts on Aztec, written in Noir. Includes useful Note implementations.
+- `noir contracts` - , example Aztec contracts
+- `noir-protocol-circuits` - a crate containing essential circuits for the protocol (public circuits and private wrappers)
+
+A lot of what we will look at will be in [aztec-nr/aztec/src/note](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note), specifically the lifecycle and note interface.
+Looking at the noir circuits in these components, you will see references to the distinction between public/private execution and state.
+
+### Lifecycle functions
+Inside the [lifecycle](
+https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/lifecycle.nr) circuits we see the functions to create and destroy a note, implemented as insertions of note hashes and nullifiers respectively.
+
+We also see a function to create a note hash from the public context, a way of creating a private variable from a public call (run in the sequencer).
+
+### Note Interface functions
+To see a [note_interface](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/note_interface.nr) implementation, we will look at a simple [ValueNote](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/value-note/src/value_note.nr).
+
+#### Computing hashes and nullifiers
+A few key functions in the note interface are around computing the note hash and nullifier, with logic to get/use secret keys from the private context.
+
+In the ValueNote implementation you'll notice that it uses the `pedersen_hash` function. This is currently required by the protocol, but may be updated to another hashing function, like poseidon.
+
+As a convenience, the outer [note/utils.nr](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/utils.nr) contains an implementation of a function required by Aztec contracts using private storage: `compute_note_hash_and_nullifier`
```

Comment:
it is no longer required. see https://github.com/AztecProtocol/aztec-packages/pull/4636

Reply:
[This](https://github.com/AztecProtocol/aztec-packages/pull/4610)?
[utils.nr](https://github.com/AztecProtocol/aztec-packages/blob/master/noir-projects/aztec-nr/aztec/src/note/utils.nr#L80) still has the function which is what is being referred to.

I'll create an issue to find and update references to `compute_note_hash_and_nullifier` in line with [4610](https://github.com/AztecProtocol/aztec-packages/pull/4610).